scipy/optimize/_differentialevolution.py

```diff
@@ -598,6 +619,43 @@ def __init__(self, func, bounds, args=(),
 
         self.random_number_generator = check_random_state(seed)
 
+        # Which parameters are going to be integers?
+        if np.any(integrality):
+            # user has provided a truth value for integer constraints
+            if polish:
+                message = ("Polishing is incompatible with integrality"
+                           " constraints; `polish` will be ignored.")
+                self.polish = False
+                warnings.warn(message, OptimizeWarning)
+
+            self.integrality = np.broadcast_to(
+                integrality,
+                self.parameter_count
+            )
```

Comment:
If you're going to use `integrality` instead of `self.integrality` anywhere below, you would want to do
```suggestion
            integrality = np.broadcast_to(
                integrality,
                self.parameter_count
            )
            self.integrality = integrality
```